### PR TITLE
Add variable decimals support

### PR DESCRIPTION
**Pull request by mihaiberq**
Use BN to format the balance of custom tokens & get rid of scientific notation

### CODE REVIEW

**Review by sneljo1 — COMMENTED** [review]
Can you elaborate? Isnt `formatUnits` meant to do this?

**Comment by mihaiberq** [comment]
might've gone overboard with this... will provide some more info soon

File: src/refinable/account/EvmAccount.ts [hunk 1]
```diff
@@ -191,4 +192,42 @@ export default class EvmAccount implements Account {
     }
     return;
   }
+
+  private formatBalance(balance: string, decimals: number) {
```
Comment (colthreepv) [inline src/refinable/account/EvmAccount.ts:196]: I am not totally sure of the intention here. you want to format a BigNumber toString?

Like `83728132.21235123517` to `"83728132.21235123517"`. Why BigNumber.from(longnumber).toString() is not enough?
https://docs.ethers.io/v5/api/utils/bignumber/#BigNumber--BigNumber--methods--conversion

Comment (mihaiberq) [reply src/refinable/account/EvmAccount.ts:196]: BigNumber doesnt support decimals, so we have to hack it

Comment (colthreepv) [reply src/refinable/account/EvmAccount.ts:196]: Oh nononono I have experience on this 😄 Let me tell you!

Comment (colthreepv) [reply src/refinable/account/EvmAccount.ts:196]: this utils function, just specify Bignumber, specify amount of decimals, and format is done
https://docs.ethers.io/v5/api/utils/display-logic/#utils-formatUnits

Comment (mihaiberq) [reply src/refinable/account/EvmAccount.ts:196]: well, i implemented that function myself it seems :D 2h learning experience lol. will revert and change the frontend code that deals with parsing to fix the issue i had

**Comment by mihaiberq** [comment]
closing this as, in the end, i managed to implement parseUnits by myself. will regard it as a learning experience.

problem lays in frontend decimal parsing, rather than this function.

**Pull request closed without merging**